src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1153,7 +1161,7 @@ public int CompareTo(object obj)
         public byte[] ToByteArray(bool isUnsigned = false, bool isBigEndian = false)
         {
             int ignored = 0;
-            return TryGetBytes(GetBytesMode.AllocateArray, default, isUnsigned, isBigEndian, ref ignored);
+            return TryGetBytes(GetBytesMode.AllocateArray, default, isUnsigned, isBigEndian, ref ignored)!;
```

Comment:
Why not instead return null or empty array?

Reply:
Not understanding the question: can you clarify?

Reply:
I mean, `TryGetBytes` can return a null `byte[]`, why did you decide to bang it away instead of making the return type for this API nullable or returning an empty array?

Reply:
Ah!

TryGetBytes has three different modes based on why it's being called: to count how much space is needed, to create an array, or to fill a span.  It will never return null when being asked to create an array, which this call site is doing with `GetBytesMode.AllocateArray`.  We just don't have a way to express that, e.g. `[return: NotNullWhen("mode", GetBytesMode.AllocateArray)]`).

Reply:
Ah, I see. Makes sense. I missed to follow the mode we where passing down at the call site. Thanks for clarifying.